portfolio/src/main/java/com/google/sps/Puzzle.java

```diff
@@ -0,0 +1,26 @@
+package com.google.sps;
+
+/** Puzzle domain object */
+public class Puzzle {
+
+  private String puzzleId; //Datastore generated id
+  private String image; // Blobstore url
```

Comment:
Nit: change the name to be more indicative of what the field is actually used for:
```suggestion
  private String imageUrl; // Blobstore url
```